src/BugsnagUnityForReview.mm

```diff
@@ -0,0 +1,876 @@
+  callback(appData, "version", [version UTF8String]);
+
+  BugsnagAppWithState *app = [[Bugsnag client] generateAppWithState:sysInfo];
+  NSString *isLaunching = app.isLaunching ? @"true" : @"false";
+  callback(appData, "isLaunching", [isLaunching UTF8String]);
+
+}
+
+void bugsnag_retrieveLastRunInfo(const void *lastRuninfo, void (*callback)(const void *instance, bool crashed, bool crashedDuringLaunch, int consecutiveLaunchCrashes)) {
+
+  int consecutiveLaunchCrashes = Bugsnag.lastRunInfo.consecutiveLaunchCrashes;
+  bool crashed = Bugsnag.lastRunInfo.crashed;
+  bool crashedDuringLaunch = Bugsnag.lastRunInfo.crashedDuringLaunch;
+
+  callback(lastRuninfo, crashed, crashedDuringLaunch, consecutiveLaunchCrashes);
+
+}
+
+void bugsnag_retrieveDeviceData(const void *deviceData, void (*callback)(const void *instance, const char *key, const char *value)) {
+  NSDictionary *sysInfo = [BSG_KSSystemInfo systemInfo];
+
+  NSFileManager *fileManager = [NSFileManager defaultManager];
+
+  NSError *error;
+  NSDictionary *fileSystemAttrs = [fileManager attributesOfFileSystemForPath:NSHomeDirectory() error:&error];
+
+  if (error) {
+      bsg_log_warn(@"Failed to read free disk space: %@", error);
+  }
+
+  NSNumber *freeBytes = [fileSystemAttrs objectForKey:NSFileSystemFreeSize];
+  callback(deviceData, "freeDisk", [[freeBytes stringValue] UTF8String]);
+
+  uint64_t freeMemory = bsg_ksmachfreeMemory();
+  char buff[30];
+  sprintf(buff, "%lld", freeMemory);
+  callback(deviceData, "freeMemory", buff);
+
+  callback(deviceData, "id", [sysInfo[@BSG_KSSystemField_DeviceAppHash] UTF8String]);
+  callback(deviceData, "jailbroken", [[sysInfo[@BSG_KSSystemField_Jailbroken] stringValue] UTF8String]);
+  callback(deviceData, "locale", [[[NSLocale currentLocale] localeIdentifier] UTF8String]);
+  callback(deviceData, "manufacturer", "Apple");
+  callback(deviceData, "model", [sysInfo[@BSG_KSSystemField_Machine] UTF8String]);
+  callback(deviceData, "modelNumber", [sysInfo[@BSG_KSSystemField_Model] UTF8String]);
+  callback(deviceData, "osName", [sysInfo[@BSG_KSSystemField_SystemName] UTF8String]);
+  callback(deviceData, "osVersion", [sysInfo[@BSG_KSSystemField_SystemVersion] UTF8String]);
+  callback(deviceData, "osBuild", [sysInfo[@BSG_KSSystemField_OSVersion] UTF8String]);
+}
+
+void bugsnag_populateUser(bugsnag_user *user) {
+  NSDictionary *sysInfo = [BSG_KSSystemInfo systemInfo];
+  user->user_id = [sysInfo[@BSG_KSSystemField_DeviceAppHash] UTF8String];
+}
+
+void bugsnag_setUser(char *userId, char *userName, char *userEmail) {
+    [Bugsnag setUser:userId == NULL ? nil : [NSString stringWithUTF8String:userId]
+            withEmail:userEmail == NULL ? nil : [NSString stringWithUTF8String:userEmail]
+             andName:userName == NULL ? nil : [NSString stringWithUTF8String:userName]];
+}
+
+@interface Bugsnag ()
++ (BugsnagNotifier *)notifier;
+@end
+
```

Comment:
This should be deleted because it declares a method that doesn't exist.

```suggestion
```

Reply:
Removed 👍